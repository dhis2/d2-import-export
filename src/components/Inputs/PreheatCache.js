import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../form/RadioGroup'

export const OPTION_YES = {
    value: 'true',
    label: i18n.t('Yes (faster for large imports)'),
}
export const OPTION_NO = { value: 'false', label: i18n.t('No') }
export const PREHEAT_CACHE_KEY = 'preheatCache'
export const PREHEAT_CACHE_DEFAULT_VALUE = OPTION_NO.value

export const PreheatCache = () => (
    <RadioGroup
        name={PREHEAT_CACHE_KEY}
        label={i18n.t('Preheat cache')}
        options={[OPTION_YES, OPTION_NO]}
    />
)
