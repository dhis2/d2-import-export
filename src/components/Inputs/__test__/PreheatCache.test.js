import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { PreheatCache } from '../PreheatCache'

jest.mock('react-final-form', () => ({
    useField: jest.fn(() => ({
        input: {
            name: 'Name',
            value: '',
            onChange: jest.fn(),
        },
        meta: {
            touched: false,
            error: '',
        },
    })),
}))

describe('Input component - PreheatCache', () => {
    it('should render correctly', () => {
        const file = shallow(<PreheatCache />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
