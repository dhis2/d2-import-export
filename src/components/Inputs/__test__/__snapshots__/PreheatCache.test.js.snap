// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - PreheatCache should render correctly 1`] = `
<RadioGroup
  dataTest="input-preheat-cache"
  label="Preheat cache"
  name="preheatCache"
  options={
    Array [
      Object {
        "label": "Yes (faster for large imports)",
        "value": "true",
      },
      Object {
        "label": "No",
        "value": "false",
      },
    ]
  }
/>
`;
